src/backend/marsha/core/api.py

```diff
@@ -73,14 +75,16 @@ def update_state(request):
     # Update the object targeted by the "object_id" and "resource_id"
     model = apps.get_model(app_label="core", model_name=key_elements["model_name"])
 
+    conditionnal_update = {}
+    if serializer.validated_data["state"] == defaults.READY:
+        conditionnal_update["uploaded_on"] = key_elements["uploaded_on"]
+        if hasattr(model, "extension"):
+            conditionnal_update["extension"] = key_elements["extension"] or ""
```

Comment:
Why not None? Could be written:
conditionnal_update["extension"] = key_elements.get("extension")

Reply:
I want an empty string saved in database. It's easier then to manipulate it.

Reply:
In fact it doesn't matter because I deal with None value in the seralizers.